clients/java/client/src/main/java/org/camunda/bpm/client/task/impl/ExternalTaskImpl.java

```diff
@@ -258,8 +259,8 @@ public <T> T getVariable(String variableName) {
 
     VariableValue variableValue = receivedVariableMap.get(variableName);
     if (variableValue != null) {
-      if(variableValue.getTypedValue() instanceof DeferredFileValue) {
-        DeferredFileValue deferredFileValue = (DeferredFileValue) variableValue.getTypedValue();
+      if(variableValue.getTypedValue() instanceof DeferredFileValueImpl) {
+        DeferredFileValueImpl deferredFileValue = (DeferredFileValueImpl) variableValue.getTypedValue();
         deferredFileValue.setExecutionId(this.executionId);
       }

```

Comment:
❓ Can't we get rid of this code now that we pass the execution id to `VariableValue`?

Reply:
No, we use this code now to pass the executionId to the variable value.